src/streams.h

```diff
@@ -296,6 +296,16 @@ class CDataStream
         data.insert(data.end(), begin(), end());
         clear();
     }
+
+    /**
+     * XOR the contents of this stream with a certain key.
+     *
+     * @param[in] key    The key used to XOR the data in this stream.
+     */
+    void Xor(const std::string& key) {
```

Comment:
To save a `%` operation for every byte and add a sanity check for empty key, I suggest:
```c++
    void Xor(const std::string& key)
    {
        assert(key.size() > 0);
        for (size_type i = 0, j = 0; i != size(); i++) {
            vch[i] ^= key[j++];
            if (j == key.size())
                j = 0;
        }
    }
```